sklearn/datasets/base.py

```diff
@@ -21,6 +21,7 @@
 from ..utils import check_random_state
 
 import numpy as np
+import pandas as pd
```

Comment:
This needs to be an optional import. Perhaps it should only be loaded locally within each function as part of the validation of `as_frame`.